src/Html2Text.php

```diff
@@ -270,15 +271,34 @@ static function iterateOverNode($node, $in_pre = false) {
 		if (isset($node->childNodes)) {
 
 			$n = $node->childNodes->item(0);
+			$previousSiblingName = null;
 
 			while($n != null) {
 
-				$text = static::iterateOverNode($n, $in_pre || $name == 'pre');
+				$text = static::iterateOverNode($n, $previousSiblingName, $in_pre || $name == 'pre', $is_office_document);
 
-				$output .= $text;
+				// Pass current node name to next child, as previousSibling does not appear to get populated
+				switch(true) {
+					case $n instanceof \DOMDocumentType:
+					case $n instanceof \DOMProcessingInstruction:
+						$previousSiblingName = null;
+						break;
+					case $n instanceof \DOMText:
```

Comment:
I'm not a fan of this magic switch/case/continue executing without the break approach, I like code that is as little magic as possible. Could you rewrite this to take out the case statement? e.g.

```
if ($n instanceof \DOMDocumentType || $n instanceof \DomProcessingInstruction) {
  $previousSiblingName = null;
} elseif ($n instanceof \DOMText && static::isWhitespace($text) {
  // continue
} else {
  $previousSiblingName = strtolower($n->nodeName);
}
```